hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -708,11 +688,11 @@ protected void readListenerConfigurations(ObjectDataInput in) throws IOException
      *                 or will be resolved to loaded classes and the actual {@code keyType} and {@code valueType} will be copied.
      *                 Otherwise, this configuration's {@code keyClassName} and {@code valueClassName} will be copied to the
      *                 target config, to be resolved at a later time.
+     * @param backupSerializationService in case serialization service isn't initialized
```

Comment:
Can you initialise the serialisation service on the config before calling this method instead of providing it as an argument?

Reply:
Here I was simply trying to resolve an NPE. I was not going to refactor this class without knowing the sideffects.
Perhaps you can ask @vbekiaris as I believe he wrote that class

Reply:
It's not a blocker but I think it would be safe to set the serialisation service before calling this method. I'm more worried about the changes in the proxy and operation system than any issues that could come up as a side effect of setting the serialisation service on CacheConfig :)